### ISSUE TITLE
hydromt: update documentation of hydroMT

### ISSUE THREAD

**Issue by rhutten** [Deltares/HYDROLIB#61]

**Comment by rhutten** [comment]
Hi @priscavdsluis,

Could you work on the TO-do's below?

**To-do's**
The navigator (nav) section in mkdocs.yml gives the structure of the documentation as it is now. 
[1] The markdown files of the HydroMT-Delft3D FM plugin are located in the folder ../hydrolib/hydromt-delft3dfm/docs instead of the normal docs folder. MKdocs cannot handle yet these relative paths. Please find a solution for this. 
[2] Make the lay-out of this tools more nice. E.g. extra bar.
[2] The api referencing does not work properly. Please check this. Could you also generate docstrings form our code in the hydrolib/hydromt-delft3dfm folder? 
[3] Inside the hydroMT-wflow documentation there is this page [model methods and components](https://deltares.github.io/hydromt_wflow/latest/user_guide/wflow_model_setup.html#). Could you try to recreate this page for the HydroMT-Delft3D FM plugin if time allows?

**branch**
We work in the branch "doc/61_update_hydromt_docs". I committed everything. 

**example page**
The [HydroMT-wflow](https://deltares.github.io/hydromt_wflow/latest/index.html) page is an example of the topics that we want to have:

**Comment by priscavdsluis** [comment]
Hi @rhutten, I have looked again once more at the API reference documentation.
There really seems to be a problem. I constantly get errors while executing a `mkdocs serve`. I am not sure if that is because our documentation (or something else) is incorrect, or if it is a bug on mkdocs side. Eitherway, I cannot find anything about the error and I tried so many things.
 
I also tried to use the mkdocs plugin "mktheapidocs" which IS able to generate the documentation with a nice link to source code on github.

```
  - mktheapidocs:
      modules: 
        hydrolib:
          section: "API reference" 
          source_repo: https://github.com/Deltares/HYDROLIB/tree/main
          hidden: ["submodules", "to", "omit"]
```

It generates good looking html files, BUT I could not find a way to do the following: 

1. Reference the generated html files so that the user can go to these pages from the home page.
2. Use the mktheapidocs plugin within the dhydamo and the hydromt_delft3dfm mkdocs.yml -> can only use it in the root mkdocs.yml. I think this is becuase the monorepo plugin does not support this...

The mktheapidocs plugin does not have any documentation on this. So I am stuck :( maybe we can sit together sometime to figure it out.
